src/Core/Types.Filters/IFilterInputTypeDescriptor.cs

```diff
@@ -111,5 +153,7 @@ IFilterInputTypeDescriptor<T> Directive<TDirective>()
         IFilterInputTypeDescriptor<T> Directive(
             NameString name,
             params ArgumentNode[] arguments);
+
+        public class RequireStruct<TStruct> where TStruct : struct { }
```

Comment:
@michaelstaib  You will probably not like this. This was the only way I found to capture the generics correctly... Part2

Reply:
Same as the above.